src/background-script/quick-and-dirty-migrations.ts

```diff
@@ -19,6 +19,58 @@ export interface Migrations {
 }
 
 export const migrations: Migrations = {
+    /*
+     * There was a bug in some annotation refactoring (due to url normalisation)
+     * that meant some annotations are created with a full url as the prefixed key
+     */
+    'normalise-all-annotation-url-keys': async ({
+        db,
+        storex,
+        normalizeUrl,
+    }) => {
+        // go through all annotation urls that beggin with http
+
+        // update their key to be normalised
+        // update their relations to be normalised (annot bookmarks, etc, )
+        // update social sharing too? yikes,
+
+        const annotationsNormalised = []
+
+        await db
+            .table('annotations')
+            .toCollection()
```

Comment:
A more efficient way to do this, doing the filtering on a lower-IIRC more efficient-level, would be be to use the Dexie `.startsWith` method afforded on a where clause on that field. Something like

`db.table('annotations').where('url').startsWith('http').modify(...)`